### PR TITLE
fix(cosmwasm): update registration flow

### PR DESCRIPTION
**Pull request by evan-gray**
This PR updates the NTT accountant registration flow to be more clear and robust. I also generated some more test coverage.

---

## Coverage Report for `handle_ntt_vaa`

Ran `cargo llvm-cov` against `ntt-global-accountant` after the added tests.                     

### Uncovered lines in `handle_ntt_vaa` (contract.rs:471–662)

Only two lines remain uncovered:
- **Line 525** — `bail!("peers are not cross-registered")` — defense-in-depth branch; unreachable via legitimate VAAs because the emitter authenticates the sender's own address and the per-`(chain, sender, peer-chain)` slot prevents asymmetric cross-registration.
- **Line 601** — `?` on `TRANSCEIVER_TO_HUB.may_load(...)` — storage-error propagation; only triggers on a `cw-multi-test` storage failure, which can't happen in tests.

Every other branch in `handle_ntt_vaa` is covered.

### File-wide coverage

| File | Lines | Covered |
|---|---|---|
| `contract.rs` | 763 | 52.82% |
| `structs/relayer.rs` | 183 | 89.07% |
| `state.rs` | 68 | 57.35% |
| `error.rs` | 12 | 0% |
| `msg.rs` | 13 | 0% |
| **TOTAL** | **1039** | **58.23%** |

The overall `contract.rs` percentage is dragged down by other unrelated top-level functions (queries, migrate, admin paths ~lines 670–954) that aren't exercised by this test suite. `handle_ntt_vaa` itself is effectively fully covered.

### Reproduce

Prerequisites

```bash
rustup component add llvm-tools-preview
cargo install cargo-llvm-cov --locked
```

```bash
cd cosmwasm
cargo llvm-cov --package ntt-global-accountant --html
# open target/llvm-cov/html/index.html
```

### CODE REVIEW

File: cosmwasm/contracts/ntt-global-accountant/src/contract.rs [hunk 1]
```diff
@@ -499,6 +499,8 @@ fn handle_ntt_vaa(
     let prefix = &payload[..4];
 
     if prefix == WormholeTransceiver::PREFIX {
+        // This is a TransceiverMessage (transfer)
```
Comment (mdulin2) [inline cosmwasm/contracts/ntt-global-accountant/src/contract.rs:502]: Thanks for adding these comments. This will save me loads of time in the future reviewing this code.

File: cosmwasm/contracts/ntt-global-accountant/src/contract.rs [hunk 2]
```diff
@@ -499,6 +499,8 @@ fn handle_ntt_vaa(
     let prefix = &payload[..4];
```
Comment (mdulin2) [inline cosmwasm/contracts/ntt-global-accountant/src/contract.rs:500]: Can we add a check that the sender and the registered peer are not on the same chain? We should check for this to prevent edge cases in the accountant keying again.

**Review by mdulin2 — DISMISSED** [review]

**Comment by johnsaigle** [comment]
@claude

File: cosmwasm/contracts/ntt-global-accountant/tests/ntt_registration.rs [hunk 3]
```diff
@@ -0,0 +1,1066 @@
...
+        SPOKE_CHAIN_A,
+        SPOKE_ADDR_A,
+        seq.next(),
+        &registration_payload(HUB_CHAIN, HUB_ADDR),
+    );
+    contract.submit_vaas(vec![vaa]).unwrap();
+
+    // spoke A has inherited the hub but is not a self-referential hub itself,
+    // so it cannot pre-register a peer that has no hub entry
+    let unknown_chain: u16 = 99;
```
Comment (johnsaigle) [inline cosmwasm/contracts/ntt-global-accountant/tests/ntt_registration.rs:947]: Can you add a case where the chain is known but the address is not, and vice-versa? Either one being incorrect should cause the same error, but right now we're checking only "both known" and "both unknown".

**Review by johnsaigle — DISMISSED** [review]

File: cosmwasm/contracts/ntt-global-accountant/tests/ntt_registration.rs (hunk 3 above)
Comment (evan-gray) [reply cosmwasm/contracts/ntt-global-accountant/tests/ntt_registration.rs:947]: https://github.com/wormhole-foundation/wormhole/pull/4756/commits/ac66e0457423bbd9a77038a95c89725398f6ff93

**Review by johnsaigle — APPROVED** [review]

**Review by panoel — APPROVED** [review]
I only looked at the file I am code owner of

**Review by mdulin2 — APPROVED** [review]
